pkg/agent/core/state.go

```diff
@@ -430,6 +478,17 @@ func (s *state) calculateNeonVMAction(
 	pluginRequested *api.Resources,
 	pluginRequestedPhase string,
 ) (*ActionNeonVMRequest, *time.Duration) {
+	targetRevision := s.TargetRevision
+	if desiredResources.HasFieldLessThan(s.VM.Using()) {
+		// We are downscaling, so we needed a permit from the monitor
+		targetRevision = targetRevision.Min(s.Monitor.CurrentRevision)
+	}
+
+	if desiredResources.HasFieldGreaterThan(s.VM.Using()) {
+		// We are upscaling, so we needed a permit from the plugin
+		targetRevision = targetRevision.Min(s.Plugin.CurrentRevision)
+	}
```

Comment:
Revisiting this — the implicit assumption is that _if_ we're downscaling/upscaling, we already have the necessary permission, right?

If so: I think this `targetRevision` logic must go after `desiredResources = s.clampResources(...)` below, but please double-check me on that (maybe, write a test case that fails without that change?)

Reply:
I'll think about this some more & report back